### PR TITLE
 Improve docstrings, a few tests.

### PR DESCRIPTION
**Pull request by jcrist**
- Add docstrings to most things
- Add tests for some untested functions
- Few small code fixes, mostly to improve consistent behavior.

### CODE REVIEW

File: datashader/core.py [hunk 1]
```diff
@@ -7,18 +7,87 @@
...
+        return type(self) is type(other) and self.inputs == other.inputs
+
+    def __ne__(self, other):
+        return not self == other
+
+
 class Axis(object):
+    """Interface for implementing axis transformations.
+
+    Instances hold implementations of implementations too and from axis space.
```
Comment (jbednar) [inline datashader/core.py]: implementations of implementations?

Presumably "to and".

File: datashader/glyphs.py [hunk 2]
```diff
@@ -2,18 +2,31 @@
...
 
 
-class Glyph(object):
+class Glyph(Expr):
+    """Base class for glyphs."""
     pass
 
 
 class Point(Glyph):
+    """A point, with center at ``x`` and ``y``.
```
Comment (jbednar) [inline datashader/glyphs.py]: An idealized (zero area) point?

I.e., can you clarify that this is the mathematical definition of point, not a dot that has a specific area (which many plotting programs will call a point)?  Later we'll add a circle, but for now people may think this is meant to be a circle.

File: datashader/reductions.py [hunk 3]
```diff
@@ -116,6 +118,7 @@ def _finalize(bases, **kwargs):
 
 
 class FloatingReduction(Reduction):
+    """Base classes for reductions that always have floating dtype."""
```
Comment (jbednar) [inline datashader/reductions.py]: "floating-point" might sound less mysterious.

File: datashader/reductions.py [hunk 4]
```diff
@@ -265,6 +327,14 @@ def _finalize(bases, **kwargs):
 
 
 class std(Reduction):
+    """Standard Deviation of all elements in ``column``.
+
+    Parameters
+    ----------
+    column : str
+        Name of the column to aggregate over. Must be numeric. ``NaN values in
+        the column are skipped.
```
Comment (jbednar) [inline datashader/reductions.py]: ``NaN``

**Comment by jbednar** [comment]
Looks great; happy to have it merged after the minor typo fixes noted.

**Comment by jbednar** [comment]
BTW, is it possible to make the tests pass before the conda package is built?  I thought it was failing due to dynd, but since that's no longer needed, it must be something else...

**Comment by jcrist** [comment]
If you click on the build failure (red x on the commit), you can see what made the test fail. In this case it was a failure to import bokeh when I added the `--doctest-modules` flag for python 2.7. All tests on master have been passing, and of course we don't merge code with tests failing.

**Pull request merged**